src/library/basesqltablemodel.cpp

```diff
@@ -293,7 +293,7 @@ void BaseSqlTableModel::select() {
     // should not disturb that if we are only removing tracks.
     qStableSort(rowInfo.begin(), rowInfo.end());
 
-    m_trackIdToRows.clear();
+    QMultiHash<TrackId, int> trackIdToRows;
```

Comment:
Switching to .clear() was something we added a long time ago to trim down the time selects took (since it reuses the hash table storage). Did you change this to avoid a race condition while populating m_trackIdToRows?